src/rdbuf.h

```diff
@@ -89,6 +89,14 @@ typedef struct rd_buf_s {
 
         rd_segment_t     *rbuf_wpos;          /**< Current write position seg */
         size_t            rbuf_len;           /**< Current (written) length */
+        size_t            rbuf_erased;        /**< Total number of bytes
+                                               *   erased from segments.
+                                               *   This amount is taken into
+                                               *   account when checking for
+                                               *   writable space which is
+                                               *   always at the end of the
+                                               *   buffer and thus can't make
+                                               *   use of the erased parts. */
```

Comment:
i haven't looked at this in depth, but want to note that it seems off to me that state related to deleting part of a buf should leak outside that scope of that operation.

Reply:
So maybe it is a good time to check what the implementation does and why this is necessary ;)